### ISSUE TITLE
E1022 rejects Fn::GetAZs inside Fn::Join, which the docs and select.json allow

### ISSUE THREAD

**Issue by zfLQ2qx2** [aws-cloudformation/cfn-lint#4674]
### CloudFormation Lint Version

1.56.0

### What operating system are you using?

Amazon Linux 2023

### Describe the bug

`Fn::GetAZs` is rejected inside `Fn::Join`'s list of values:

```
E1022 {'Fn::GetAZs': ''} is not of type 'array'
```

The CloudFormation Template Reference for `Fn::Join`, under "Supported
functions", lists `Fn::GetAZs` among the functions allowed in the list of values.
https://docs.aws.amazon.com/AWSCloudFormation/latest/TemplateReference/intrinsic-function-reference-join.html

`cfnlint/data/schemas/other/functions/join.json` allows this in the list
argument:

```json
"functions": ["Fn::Cidr", "Fn::FindInMap", "Fn::GetAtt", "Fn::If", "Fn::Split", "Ref"]
```

`select.json`, for the list argument of `Fn::Select`, has the identical list plus
`Fn::GetAZs`, which is why `!Select [0, !GetAZs '']` passes and the `Fn::Join`
form of the same idiom does not.

Real use is `AvailabilityZones: !Join [',', !GetAZs '']` on an
`AWS::ElastiCache::CacheCluster`, where the property wants a comma-delimited
string of every AZ in the region.

### Expected behavior

No E1022. `Fn::GetAZs` added to the list-argument functions in `join.json`,
matching `select.json` and the documented list.

### Reproduction template

```yaml
AWSTemplateFormatVersion: '2010-09-09'
Resources:
  A:
    Type: 'Custom::Thing'
    Properties:
      ServiceToken: 'arn:aws:sns:us-east-1:123456789012:t'
      One: !Select [0, !GetAZs '']
      Two: !Join [',', !GetAZs '']
      Three: !Join [',', !Split [',', 'a,b']]
```

```
$ cfn-lint select.yaml
select.yaml:8:24:8:34:E1022:{'Fn::GetAZs': ''} is not of type 'array'
```

Only the `Fn::Join` line is rejected.
